Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/ArgumentSpace.java

```diff
@@ -0,0 +1,373 @@
+            
+            if (i > 0) {
+                Region prev = leftArguments.get(i-1);
+                // The i-th (left)argument is placed to the right of argument i-1, with a horizontal space of H_GAP between them.
+                arg.layoutXProperty().bind(prev.layoutXProperty().add(prev.widthProperty()).add(H_GAP));
+            }            
+            this.getChildren().add(arg);
+        }
+        
+        //Put rightArgument's drawOrder on top 
+        rightArgument.toFront();
+        //Place rightArgument to the right of the knot, with a horizontal space of H_GAP between them.
+        rightArgument.layoutXProperty().bind(knot.layoutXProperty().add(knot.radiusProperty()).add(H_GAP));
+        centerLayoutVertical(rightArgument);
+        
+        //Put knot's drawOrder on to (above rightArgument)
+        knot.toFront();
+        //Vertically center the knot.
+        knot.layoutYProperty().bind(this.heightProperty().divide(2));
+
+        //Mouse listeners
+        knot.setOnMousePressed(event -> {knotPressed(ConnectionCreationManager.INPUT_ID_MOUSE); event.consume();});
+        knot.setOnMouseDragged(event -> {knotMoved(event); event.consume();});
+        knot.setOnMouseReleased(event -> {knotReleased(ConnectionCreationManager.INPUT_ID_MOUSE); event.consume();});
+        
+        //Touch listeners
+        knot.setOnTouchPressed(event -> {knotPressed(event.getTouchPoint().getId()); event.consume();});
+        knot.setOnTouchMoved(event -> {knotMoved(event); event.consume();});
+        knot.setOnTouchReleased(event -> {knotReleased(event.getTouchPoint().getId()); event.consume();});
+        
+        //Update the size of this Pane
+        this.setPrefHeight(HEIGHT);
+        this.setMaxHeight(USE_PREF_SIZE);
+        this.setMinWidth(USE_PREF_SIZE);
+        this.setMaxWidth(USE_PREF_SIZE);
+        
+        this.prefWidthProperty().bind(getTotalWidthProperty());
+        
+        // Since at the point of a layout update the width of the Labels is unknown, we have to ask for another layout pass.
+        rightArgument.widthProperty().addListener(a -> Platform.runLater(block::updateLayout));
+        
+        snapToKnotIndex();
+    }
+    
+    /**
+     * @return DoubleBinding that represents the entire width of all the elements represented in this ArgumentSpace.
+     */
+    public DoubleBinding getTotalWidthProperty() {
+        return rightArgument.layoutXProperty().add(rightArgument.translateXProperty()).add(rightArgument.widthProperty()).add(H_GAP);
+    }
+    
+    /**
+     * Binds a region's layoutY property to always be vertically centered in the ArgumentSpace. 
+     * @param region
+     */
+    public void centerLayoutVertical(Region region) {
+        region.layoutYProperty().bind(this.heightProperty().divide(2).subtract(region.heightProperty().divide(2)));
+    }    
+
+    /**
+     * @return The knot's position used to determine the knot's index.
+     */
+    public double getKnotPos() {
```

Comment:
Does this need to be a public method? It seems you only use it in `determineKnotIndex`.